tests/virt_autotest/validate_system_health.pm

```diff
@@ -0,0 +1,84 @@
+# SUSE's openQA tests
+#
+# Copyright 2012-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Do a basic examination to the host and guests after tests run
+# Maintainer: Julie CAO <JCao@suse.com>, qe-virt@suse.de
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use ipmi_backend_utils qw(reconnect_when_ssh_console_broken);
+use Utils::Architectures;
+use virt_autotest::common;
+use virt_autotest::utils qw(is_xen_host start_guests check_port_state check_host_health check_guest_health);
+use virt_utils qw(collect_host_and_guest_logs);
+
+sub prepare_run_test {
+    reconnect_when_ssh_console_broken unless defined(script_run("rm -f /root/{commands_history,commands_failure}", die_on_timeout => 0));
+    script_run("history -c");
+}
+
+sub run_test {
+
+    my $self = shift;
+    my %health_status = ();
+    $health_status{host} = check_host_health;
+    foreach my $guest (keys %virt_autotest::common::guests) {
```

Comment:
I think 
```
@healthy_status{(keys %virt_autotest::common::guests)} = ();
foreach my $guest (grep { $_ ne 'host' } keys %health_status) {
```
is better. Because it only involves healthy_status in function main body, get rid of dependency on global virt_autotest::common::guests.

Reply:
I think it is shorter. If you are afraid of global dependency, I can create an array for `keys %virt_autotest::common::guests` at the begging the way Tony did.

Reply:
If you think you can do it, then do it. I mean if you think that will make the test more clean and clear and well structured.